eo-maven-plugin/src/main/java/org/eolang/maven/ParseMojo.java

```diff
@@ -104,85 +98,81 @@ public final class ParseMojo extends SafeMojo {
-                                            "Unable to parse %s",
-                                            tojo.get(Tojos.KEY)
-                                        ),
-                                        ex
-                                    );
-                                }
-                            }
-                        )
-                    );
-                }
-            );
-        try {
-            for (final Future<Object> completed : Executors.newFixedThreadPool(this.threads)
-                .invokeAll(tasks)) {
-                try {
-                    completed.get();
-                } catch (final ExecutionException ex) {
-                    throw new IllegalArgumentException(
-                        ex.getCause().getMessage(),
-                        ex
-                    );
-                }
-            }
-        } catch (final InterruptedException ex) {
-            Thread.currentThread().interrupt();
-            throw new IllegalStateException(
-                String.format(
-                    "Interrupted while waiting for %d parsing to finish",
-                    total.get()
-                ),
-                ex
-            );
-        }
-        if (total.get() == 0) {
-            if (tojos.isEmpty()) {
+        final int total = tasks
+            .parallelStream()
+            .mapToInt(Supplier::get)
+            .sum();
+        if (0 == total) {
+            if (((Collection<Tojo>) this.scopedTojos().select(
+                row -> row.exists(AssembleMojo.ATTR_EO)
+            )).isEmpty()) {
                 Logger.info(this, "No .eo sources need to be parsed to XMIRs");
             } else {
                 Logger.info(this, "No .eo sources parsed to XMIRs");
             }
         } else {
-            Logger.info(this, "Parsed %d .eo sources to XMIRs", total.get());
+            Logger.info(this, "Parsed %d .eo sources to XMIRs", total);
+        }
+    }
+
+    /**
+     * Create a task for parsing a single EO file.
+     *
+     * @param tojo Tojo
+     * @return Task
+     */
+    private Supplier<Integer> task(final Tojo tojo) {
+        final ClassLoader loader = Thread.currentThread().getContextClassLoader();
+        return () -> {
+            try {
+                Thread.currentThread().setContextClassLoader(loader);
+                this.parse(tojo);
+                return 1;
+            } catch (final IOException ex) {
+                throw new IllegalStateException(
+                    String.format(
+                        "Unable to parse %s",
+                        tojo.get(Tojos.KEY)
+                    ),
+                    ex
+                );
+            }
+        };
+    }
+
+    /**
+     * Check if the given tojo has already been parsed.
+     *
+     * @param tojo Tojo.
+     * @return True if the tojo has already been parsed.
+     */
+    private boolean hasNotAlreadyParsed(final Tojo tojo) {
```

Comment:
@volodya-lombrozo let's make the name simpler: `isParsed`?

Reply:
@mximp done